modules/libcom/src/osi/os/posix/osdThread.c

```diff
@@ -111,44 +112,91 @@ if(status) { \
     exit(-1);\
 }
 
-
-epicsShareFunc int epicsThreadGetPosixPriority(epicsThreadId pthreadInfo)
-{
 #if defined(_POSIX_THREAD_PRIORITY_SCHEDULING) && _POSIX_THREAD_PRIORITY_SCHEDULING > 0
-    double maxPriority,minPriority,slope,oss;
+static int osi2posixPriority(unsigned osiPriority)
+{
+    double maxPriority,minPriority,slope;
+    int    oss;
+
+    if (!pcommonAttr->valid)
+        return 0; /* assume SCHED_OTHER is in effect */
 
     if(pcommonAttr->maxPriority==pcommonAttr->minPriority)
-        return(pcommonAttr->maxPriority);
+        return(pcommonAttr->minPriority);
+
     maxPriority = (double)pcommonAttr->maxPriority;
     minPriority = (double)pcommonAttr->minPriority;
-    slope = (maxPriority - minPriority)/100.0;
-    oss = (double)pthreadInfo->osiPriority * slope + minPriority;
-    return((int)oss);
+    slope = (double)(maxPriority - minPriority)/(double)(epicsThreadPriorityMax - epicsThreadPriorityMin);
+    oss   = (int)round( (double)(osiPriority - epicsThreadPriorityMin) * slope + minPriority );
+    if ( oss < pcommonAttr->minPriority )
+        oss = pcommonAttr->minPriority;
+    if ( oss > pcommonAttr->maxPriority )
+        oss = pcommonAttr->maxPriority;
+    return oss;
+}
+
+static unsigned posix2osiPriority(int posixPriority)
+{
+double   slope, osid;
+unsigned osi;
+
+    if ( ! pcommonAttr->valid )
+        return epicsThreadPriorityMedium;
+
+    if ( pcommonAttr->maxPriority == pcommonAttr->minPriority )
+        return epicsThreadPriorityMedium;
+
+    slope =   (double)(epicsThreadPriorityMax   - epicsThreadPriorityMin)
+            / (double)(pcommonAttr->maxPriority - pcommonAttr->minPriority);
```

Comment:
That's not known. The sysadmin can set a ceiling for the RT-priority the process may acquire (RLIMIT_RTPRIO). This is also why the available range of OSS priorities is computed dynamically ('findPriorityRange'). It is therefore possible that the process is allowed to use SCHED_FIFO but it may only use a single priority level. This is why this test is needed.

Reply:
Maybe the possibility of divide by zero needs to be handled then?

Reply:
Hmm  - I think it's pretty obvious that this *is* actually handled.

````
if ( a == b ) {
  return something;
}
/* a-b cannot be zero here */
y = x/(a-b);
````

Reply:
Ok, my mistake.  I was focused too much on the logic in `once()`.